web3/contract.py

```diff
@@ -190,6 +185,8 @@ def __init__(self,
 
         if address is not empty:
             self.address = address
+            self._validate_address()
+            self._normalize_address()
```

Comment:
Can you instead extract the validation and normalization logic to be stand-alone functions rather than additional methods on this object.  Then you can just change this to be `self.address = normalize_address(address)`.